src/modules/job-exec/exec_config.c

```diff
@@ -283,14 +279,61 @@ int config_init (flux_t *h, int argc, char **argv)
         /* Finally, override values on cmdline */
         for (int i = 0; i < argc; i++) {
             if (strstarts (argv[i], "job-shell="))
-                exec_conf.default_job_shell = argv[i]+10;
+                tmpconf.default_job_shell = argv[i]+10;
             else if (strstarts (argv[i], "imp="))
-                exec_conf.flux_imp_path = argv[i]+4;
+                tmpconf.flux_imp_path = argv[i]+4;
             else if (strstarts (argv[i], "service="))
-                exec_conf.exec_service = argv[i]+8;
+                tmpconf.exec_service = argv[i]+8;
         }
     }
 
+    exec_conf.default_job_shell = tmpconf.default_job_shell;
+    exec_conf.flux_imp_path = tmpconf.flux_imp_path;
+    exec_conf.exec_service = tmpconf.exec_service;
+    exec_conf.exec_service_override = tmpconf.exec_service_override;
+    exec_conf.sdexec_properties = tmpconf.sdexec_properties;
+    return 0;
+}
+
+/*  Initialize common configurations for use by job-exec exec modules.
+ */
+int config_init (flux_t *h, int argc, char **argv)
+{
+    const flux_conf_t *conf;
+    flux_error_t err;
+
+    /* Per trws comment in 97421e88987535260b10d6a19551cea625f26ce4
+     *
+     * The musl libc loader doesn't actually unload objects on
+     * dlclose, so a subsequent dlopen doesn't re-clear globals and
+     * similar.
+     *
+     * So we must re-initialize globals everytime we reload the module.
+     * This will be handled in conf_parse().
+     */
+
+    if (!(conf = flux_get_conf (h))) {
+        flux_log_error (h, "error retrieving flux conf");
+        return -1;
+    }
+
+    if (conf_parse (h, conf, argc, argv, &err) < 0) {
+        flux_log (h, LOG_ERR, "%s", err.text);
+        return -1;
+    }
+
+    return 0;
+}
+
+int config_reconfig (flux_t *h,
+                     const flux_conf_t *conf,
+                     int argc,
+                     char **argv,
+                     flux_error_t *errp)
+{
+    if (conf_parse (h, conf, argc, argv, errp) < 0)
+        return -1;
```

Comment:
Again, if you stash the conf in the handle as in `job-manager` and `resource`, maybe this function could go away?

Reply:
perhaps i'm missing something, but it seems that in `job-manager` and `resource`, they stash the conf in the handle **after** it has been parsed.  My assumption is that if the new conf is bad/invalid in some way, you don't want to save it?

Reply:
That said, it doesn't mean can't share the same function for config, may it be init or reload.  Maybe a general `config_setup()` or `config_parse()` callback.

The only question in my mind is that design wise, could there be a exec implementation in the future that needs a separate init & reload?

Reply:
actually, if we have a shared function, the struct is no longer needed either ... lemme re-work